decompiler/extractor/main.cpp

```diff
@@ -407,6 +432,25 @@ int main(int argc, char** argv) {
       }
       path_to_iso_files = data_dir_path;
     }
+
+    // write out a json file with some metadata for the game
+    std::string buildinfo_json = "{\n";
+    buildinfo_json += "  \"flags\": [";
+    bool first = true;
+    for (const auto& [n, f] : sGameIsoFlagNames) {
+      if (flags & f) {
+        if (first) {
+          first = false;
+        } else {
+          buildinfo_json += ", ";
+        }
+        buildinfo_json += fmt::format("\"{}\"", n);
+      }
+    }
+    buildinfo_json += "]\n";
+    buildinfo_json += "}\n";
```

Comment:
the json library can write json too. You could do something like:
```
nlohmann::json json;
auto& flags = json["flags"];
flags.push_back("foo");
...
json.dump();
```
if you want. (I haven't actually checked if this compiles)

Reply:
yeah, let's avoid hand-crafting JSON strings

Reply:
This should produce `[]`:
```
nlohmann::json json = nlohmann::json::array();
```

Reply:
I'm not sure how to add that to an existing json object